src/cache/InvalidationPolicyCache.ts

```diff
@@ -380,6 +381,38 @@ export default class InvalidationPolicyCache extends InMemoryCache {
     return this._expire(true);
   }
 
+  readEntity(id: string) {
+    return this.entityTypeMap.readEntityById(id);
+  }
+
+  readEntityExpired(id: string) {
+    const entity = this._expire(true).find(
+      (expiredEntityId) => expiredEntityId === id
+    );
+
+    return Boolean(entity);
+  }
+
+  readEntityExpirationTime(id: string) {
+    const entity = this.entityTypeMap.readEntityById(id);
+
+    if (!entity) {
+      return null;
+    }
+
+    const { cacheTime, typename } = entity;
+
+    const timeToLive =
+      this.invalidationPolicies?.types?.[typename]?.timeToLive ||
```

Comment:
Could we make a helper function for calculating the expiration time to be reused here: https://github.com/NerdWalletOSS/apollo-cache-policies/blob/4bb696ee073d4b31e99c2ff37a2beeaa3054910e/src/policies/InvalidationPolicyManager.ts#L227

and add that helper as a member to the InvalidationPolicyManager class that the cache here could then use like `this.invalidationPolicyManager.readEntityExpirationTime(id)`?

Reply:
Hey @danReynolds, sorry, a little confused on the above. If possible can you add more details here just to make sure I maket he correct change?

Reply:
Hey just got back from holiday. Yep so I'm thinking we'd add a function called readEntityExpirationTime to the invalidation policy manager class: https://github.com/NerdWalletOSS/apollo-cache-policies/blob/4bb696ee073d4b31e99c2ff37a2beeaa3054910e/src/policies/InvalidationPolicyManager.ts#L182

It would look like this:

```
readEntityExpirationTime(id: string) {
  const { entityTypeMap, policies } = this.config;

  // This simple helper function would also be added to the entity type map class
  const cacheTime = entityTypeMap.readEntityCacheTime(id);

  if (!cacheTime) {
    return null;
  }

  const timeToLive =
      this.getPolicy(typename)?.timeToLive || policies.timeToLive;

  return entityCacheTime + timeToLive;
}
```